feature/refactor-versions

puwun opened the pull request:
## Description
This PR introduces a centralized `VERSION` file to serve as the single source of truth for version management across the Rogue project. All components now read the version from this file, removing the need to manually update version numbers in multiple places.

<!-- Provide a clear and concise description of your changes -->

## Motivation and Context
Previously, version numbers were hardcoded in three different files:
- `packages/tui/internal/tui/app.go` 
- `rogue/__init__.py` 
- `pyproject.toml` 

This led to unnecessary maintenance overhead and inconsistencies.

<!-- Why is this change required? What problem does it solve? -->
<!-- If it fixes an open issue, please link to the issue here using "Fixes #123" -->

## Type of Change

<!-- Please delete options that are not relevant and check the one that applies -->


- [x] ✨ New feature (non-breaking change which adds functionality)
- [x] 🔧 Configuration/build changes

## Changes Made

- Added `VERSION` file as single source of truth containing version `0.1.12`
- Updated `rogue/__init__.py` to read version from VERSION file at runtime with fallback to package metadata
- Updated `packages/tui/internal/tui/version.go` to use build-time variable injection instead of hardcoded value
- Updated `packages/tui/Makefile` to read VERSION file and inject version via ldflags
- Updated `packages/tui/cmd/rogue/main.go` to add `version` command and `--version` flag
- Updated `pyproject.toml` to use hatch version plugin with VERSION file
- Removed hardcoded version strings from multiple locations


## Screenshots/Examples 
<img width="2738" height="1001" alt="image" src="https://github.com/user-attachments/assets/f713cb84-88ee-40da-ac9a-5ed848110c03" />

<img width="1835" height="718" alt="image" src="https://github.com/user-attachments/assets/2cd5c7b7-c2d0-4fe0-af7e-bc619b99e4ba" />

<!-- If your changes affect the UI or behavior, please provide screenshots or examples -->

## Checklist

<!-- Please check all applicable items. You can check them after creating the PR if needed -->

- [x] I have read the [CONTRIBUTING.md](../CONTRIBUTING.md) guide
- [x] My code follows the code style of this project (PEP 8, type hints, docstrings)
- [x] I have run `uv run black .` to format my code
- [x] I have run `uv run flake8 .` and fixed all issues
- [x] I have run `uv run mypy --config-file .mypy.ini .` and addressed type checking issues
- [x] I have run `uv run bandit -c .bandit.yaml -r .` for security checks
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have run `uv run pytest` and all tests pass
- [x] I have manually tested my changes
- [x] I have updated the documentation accordingly
- [x] I have added/updated type hints for new/modified functions
- [x] My changes generate no new warnings
- [x] I have checked my code for security issues
- [x] Any dependent changes have been merged and published



## Related Issues/PRs

<!-- Link to related issues or PRs -->

- Fixes #83 

**Test Steps**:
```bash
# Python package version
$ uv run python -c "import rogue; print(rogue.__version__)"
0.1.12

# TUI version command
$ cd packages/tui && make build && ./dist/rogue-tui version
rogue-tui v0.1.12

# TUI version flag
$ ./dist/rogue-tui --version
rogue-tui v0.1.12
```

puwun commented:
heyy @yuval-qf, I noticed `sdks/python/pyproject.toml` has a hardcoded version of `0.1.4`, while the main project is at `0.1.12` 
my plan is to align the sdk with the main version logic by making its version dynamic and pointing it to the root `VERSION` file,
this would change the SDK's version to 0.1.12 as well

does this sound correct to you? just wanted to double-check before I push the changes.

yuval-qf reviewed (commented):
Thanks @puwun for this PR, left a few minor comments

puwun commented:
@yuval-qf , i have centralized versioning so both **rogue-ai** and **rogue-ai-sdk** use the single root `VERSION` file

So, the key challenge was that the rogue-ai-sdk is a standalone package and can't access the project's file structure after a user installs it. To solve this, I now have embedded the version number into the sdk's package metadata during the build process itself. 
The sdk then reads this metadata at runtime to get its version

To improve the local development workflow, I've also updated the root `pyproject.toml` : 

```bash
[tool.uv.sources]
rogue-ai-sdk = { path = "sdks/python", editable = true }
```

This tells our tools to use the local sdk code instead of downloading it from a repository, this gives us a true monorepo workflow where changes to the sdk are instantly available to the main application, which speeds up development and testing 🚀🚀

I also took care of the Go code formatting issue in `main.go` and addressed the bandit security warning 

Pls take a look and suggest if any changes needed

yuval-qf commented:
> @yuval-qf , i have centralized versioning so both **rogue-ai** and **rogue-ai-sdk** use the single root `VERSION` file
> 
> So, the key challenge was that the rogue-ai-sdk is a standalone package and can't access the project's file structure after a user installs it. To solve this, I now have embedded the version number into the sdk's package metadata during the build process itself. The sdk then reads this metadata at runtime to get its version
> 
> To improve the local development workflow, I've also updated the root `pyproject.toml` :
> 
> ```shell
> [tool.uv.sources]
> rogue-ai-sdk = { path = "sdks/python", editable = true }
> ```
> 
> This tells our tools to use the local sdk code instead of downloading it from a repository, this gives us a true monorepo workflow where changes to the sdk are instantly available to the main application, which speeds up development and testing 🚀🚀
> 
> I also took care of the Go code formatting issue in `main.go` and addressed the bandit security warning
> 
> Pls take a look and suggest if any changes needed

Thanks @puwun, this solution seems great!

It seems that the `pr/codestyle` GitHub action had failed due to some formatting issues. Could you please address them?

yuval-qf commented on sdks/python/rogue_sdk/__init__.py:
```diff
@@ -51,9 +51,20 @@ async def main():
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    pass
```
```suggestion
```

yuval-qf reviewed (approved):
Thanks @puwun for this first community contribution!
Happy to merge this :+1:
